### PR TITLE
185905983-better-apollo-error-message

### PR DESCRIPTION
**Pull request by ttoomey**
## Description
https://www.pivotaltracker.com/story/show/185905983

* add reload button to snack bar
* show improved message for network vs application / server errors
* hide omni search popover when error occurs so it doesn't obscure the snack bar message

## Type of change
- [x] New feature (adds functionality)
- [X] Bugfix

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: src/modules/errors/components/ApolloErrorAlert.tsx [hunk 1]
```diff
@@ -8,40 +8,61 @@ import {
...
 const BaseAlert = forwardRef<Props2, any>(
-  ({ alertProps, retry, graphqlErrors }, ref) => {
+  ({ alertProps, retry, errors, isNetworkError }, ref) => {
+    let errorMessage = '';
+    if (isNetworkError) {
+      errorMessage = `There was a problem connecting to the network. Please reload the page and try again.`;
+    } else {
+      errorMessage = errors[0]?.message;
+      // replace terse unknown error message with something more helpful
+      if (!errorMessage || errorMessage == UNKNOWN_ERROR_HEADING) {
```
Comment (gigxz) [inline src/modules/errors/components/ApolloErrorAlert.tsx]: This is dependent on the fact that `UNKNOWN_ERROR_HEADING` == [INTERNAL_ERROR_DISPLAY_MESSAGE on the backend](https://github.com/greenriver/hmis-warehouse/blob/58843a72e2db8b1e864ba253d9b0c9bb8d692d32/drivers/hmis/lib/hmis_errors/api_error.rb#L10)?

I was thinking that the backend itself would return the user-facing error message, which is what happens currently. Feels a bit shaky to rely on the strings being equal.

**Comment by gigxz** [comment]
@ttoomey this is a bit of scope creep, but a couple suggestions:
* can we update the [error fallback](https://github.com/greenriver/hmis-frontend/blob/ffa8864467597bb4cc154e58310555433236d6d6/src/modules/errors/components/ErrorFallback.tsx#L11) to show the "contact your administrator" message as well? at this point I don't know if we can access the JSON error from the backend anymore (if its a backend error), but maybe it comes through the args.
* ~~can we update [generic table with data](https://github.com/greenriver/hmis-frontend/blob/ffa8864467597bb4cc154e58310555433236d6d6/src/modules/dataFetching/components/GenericTableWithData.tsx#L203) so it also shows an error toast if the query failed? currently it's just throwing the error which is then caught by an error boundary. there are several other places where we throw as well, but the generic table is just SUCH a widely used component, I think its worth having it's own error display (As opposed to falling back to the full page msg).~~
  * EDIT: hmm, nevermind, I think I may have been confused. It already gets wrapped in its own error boundary, so it shouldn't be doing the full-page thing. let me see why I was seeing that.

LMK if you'd like to discuss!

File: src/modules/errors/components/ApolloErrorAlert.tsx [hunk 2]
```diff
@@ -81,20 +102,35 @@ const ApolloErrorAlert: React.FC<Props> = ({
...
+    ? error.graphQLErrors?.length == 0 && !isServerError(error.networkError)
+    : false;
   if (!error) return;
 
   if (inline) {
-    return <BaseAlert {...props} graphqlErrors={graphqlErrors} />;
+    return (
+      <BaseAlert
+        {...props}
+        graphqlErrors={errors || []}
```
Comment (gigxz) [inline src/modules/errors/components/ApolloErrorAlert.tsx]:
```suggestion
        errors={errors || []}
```

File: src/modules/errors/components/ApolloErrorAlert.tsx (hunk 1 above)
Comment (ttoomey) [reply src/modules/errors/components/ApolloErrorAlert.tsx]: definitely, I'll pull this into the backend for now. 

Capturing our slack convo: in the future we might consider giving the error object a type and allow the FE to handle the display

**Comment by ttoomey** [comment]
> @ttoomey this is a bit of scope creep, but a couple suggestions:
> 
> * can we update the error fallback

Good catch, I updated the language on the error fallback. Also adjusted treatment a bit and added a reload button.

> * ~can we update so it also shows an error toast if the query failed? currently it's just throwing the error

Fixed this error. Thanks for finding it!

**Review by gigxz — APPROVED** [review]
Looks good, thanks!!

**Pull request merged**